buildmodel/commands.go

```diff
@@ -319,6 +318,26 @@ func SubmitUpdatePR(f *PRFlags) error {
 	return nil
 }
 
+// GetTempPathInDir creates a temporary path inside the given root dir. Turns local time into a
+// sortable path-friendly name to use as the name. This allows a command to run multiple times in
+// sequence without overwriting or deleting the old data, for diagnostic purposes. This is intended
+// as a dev convenience, not to enable running in parallel.
+func GetTempPathInDir(rootDir string) (string, error) {
```

Comment:
I think it makes it safer overall, too, since it actually guarantees it won't pick the same directory. Can also remove some of the timestamp's precision since it's only useful for manually poking through these directories at that point.

Reply:
I didn't think to test this change in a clean environment, and it ended up failing in the pipeline:

https://dev.azure.com/dnceng/internal/_build/results?buildId=1455046&view=logs&j=4a0cec82-aaa7-5c07-9c1e-30cc88945a51&t=d35104de-7d40-5aa7-a628-491fd52b294b&l=13

```
stat /home/vsts/work/1/s/eng/artifacts/sync-upstream-temp-repo: no such file or directory
```

I think it's because `MkdirTemp` actually tries to create the directory, but at this point, the parent directory doesn't exist yet. It used to be that `git init {temp path}` was the first place the result of this func gets used, and git handles creating parent directories if necessary. `MkdirTemp` doesn't seem to.

My initial plan to fix it is to try to find a simple way to create any necessary chain of parent directories in Go. Working on it....

Reply:
* Submitted https://github.com/microsoft/go-infra/pull/10